File: Week6/Day3/Solutions/C/prog2.c
int printf(const char*,...);
int scanf(const char*,...);
#include<math.h>

void main() {
	int num,rem,a[5],i,val,val1,length=0;
	scanf("%d",&num);
	val1 = num;
	while(val1){
		length+=1;
		val1/=10;
	}
	while(num){
		rem = num/pow(10,length-1);
		num %= (int)pow(10,length-1);
		length-=1;
		i=0;
		val=rem;
		while(rem){
			a[i]=rem%2;
			rem/=2;
			i++;
		}
		printf("Binary of %d: ",val);
		for(int j=i-1;j>=0;j--)
			printf("%d",a[j]);
		printf("\n");
	}
}

File: Week6/Day3/Solutions/Python/prog2.py
num = int(input())
length = len(str(num))
while num:
    rem = num//10**(length-1)
    num %= 10**(length-1)
    length -= 1
    ans = ""
    if rem==1:
        print("Binary of",rem,"= 000"+str(rem))
    val=rem
    if rem>1 and rem<=3:
        while rem:
            ans += str(rem%2)
            rem //= 2
        print("Binary of",val,"= 00"+ans[::-1])
    elif rem>3 and rem<=7:
        while rem:
            ans += str(rem%2)
            rem //= 2
        print("Binary of",val,"= 0"+ans[::-1])

    else:
        while rem:
            ans += str(rem%2)
            rem //= 2
        print("Binary of",val,"=",ans[::-1])
